pkg/ruler/ruler.go

```diff
@@ -602,8 +592,21 @@ func (r *Ruler) listRuleGroupsToSyncSharded(ctx context.Context, reason rulesSyn
 		})
 	}
 
-	err = g.Wait()
-	return result, err
+	// Wait until all the rule groups have been loaded.
+	err := g.Wait()
+	if err != nil {
+		return nil, err
+	}
+
+	// Filter out users which have been explicitly disabled.
```

Comment:
We should do this step before loading rules for such users (in the `for` loop at the top of this method).

Reply:
Agree. Let me do it in a follow up PR, since it's more an enhancement than a refactoring.